goosebit/ui/bff/software/routes.py

```diff
@@ -23,3 +33,72 @@ def search_filter(search_value):
     total_records = await Software.all().count()
 
     return await BFFSoftwareResponse.convert(request, query, search_filter, total_records)
+
+
+@router.delete(
+    "",
+    dependencies=[Security(validate_user_permissions, scopes=["software.delete"])],
+)
+async def software_delete(_: Request, files: SoftwareDeleteRequest) -> StatusResponse:
```

Comment:
Code being duplicated. As long as API stays the same It can simply be forwarded?

Reply:
My only thought here was if we intend to change the API at any point before 1.0.0 this will need to be changed anyway, and I think just leaving the code duplicated and separate might save some mistakes later.  I'm all for just calling one endpoint from the other, but in which direction?  Probably calling BFF from API makes the most sense, if API changes we just change it there, but if BFF changes it's a breaking API change.

Reply:
I'd done it the other way around: as we should keep the v1 API stable (as external scripts will start using it), I'd added the code there. As long as BFF is happy with that code it can re-use it - if it wants something different, then it has to implement it on its own. 

Breaking changes in BFF are not problematic, as server and client are under our direct control. Breaking changes in the API should not happen, it requires to introduce a v2.